core/types/src/lib.rs

```diff
@@ -23,11 +23,27 @@ pub use crate::{
     attestation_key::{AttestationKeyId, ExtendedAttestationKeyId},
     attributes::{Attributes, MiscellaneousAttribute, MiscellaneousSelect},
     config_id::ConfigId,
-    error::{Error, FfiError, Result},
+    error::{Error, FfiError},
     key_request::{KeyName, KeyPolicy, KeyRequest, KeyRequestBuilder},
     measurement::{Measurement, MrEnclave, MrSigner},
     quote::QuoteNonce,
     report::{Report, ReportBody, ReportData},
     svn::{ConfigSvn, CpuSvn, IsvSvn},
     target_info::TargetInfo,
 };
+
+// For targets that don't have a random number source we force it to always
+// fail.
+// Per https://docs.rs/getrandom/latest/getrandom/macro.register_custom_getrandom.html
+// this function will *only* be used if getrandom doesn't know of a native
+// secure spng
+#[cfg(target_os = "none")]
+use getrandom::register_custom_getrandom;
+
+#[cfg(target_os = "none")]
+register_custom_getrandom!(always_fail);
```

Comment:
This macro choked on the crate specific `Result`, since there was only one use of it, I removed it.